src/app/services/media-to-grid-media.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+import { GridMediaItem } from '../types/grid-media-item';
+import { MediaType } from '../types/media-type';
+import { Movie } from '../types/movie';
+import { Person } from '../types/person';
+import { TvShow } from '../types/tv-show';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MediaToGridMediaService {
+  // TODO: Ask how to get rid of switch and use factory
```

Comment:
Sorry, I couldn't find a good way to using factory. Typescript couldn't resolve type from map object. So I think it's okay to use switch/case